File: C2304L/DMA/ex001/Dtos/Responses/UserResponse.cs
using System.ComponentModel.DataAnnotations;
using ex001.Models;
namespace ex001.Dtos.Responses
{
    public class UserResponse
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string FullName { get; set; }
        public string Role { get; set; }
        public static UserResponse FromUser(User user) {
            return new UserResponse
            {
                Id = user.Id,
                Email = user.Email,
                FullName = user.FullName,
                Role = user.Role,
            };
        }

    }
}

File: C2206LDoiCan/IASF/de09/src/main/java/com/aptech/de09/dtos/responses/UserResponse.java
package com.aptech.de09.dtos.responses;

import com.aptech.de09.models.User;
import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponse {
    private Long id;

    private String username;

    private String email;

    private Date createdAt;

    private Date updatedAt;
    //factory method
    public static UserResponse fromUser(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .email(user.getEmail())
                .createdAt(user.getCreatedAt())
                .username(user.getUsername())
                .updatedAt(user.getUpdatedAt())
                .build();
    }
}
